### PR TITLE
Fix a thinko in the CallSite handling code:

### PR DESCRIPTION
**Pull request by jimingham**
I have to check for the sc list size being changed by the call-site search, not just that it had more than one element.

Added a test for multiple CU's with the same name in a given module, which would have caught this mistake.

We were also doing all the work to find call sites when the found decl and specified decl's only difference was a column, but the incoming specification hadn't specified a column (column number == 0).

### CODE REVIEW

File: lldb/test/API/functionalities/breakpoint/same_cu_name/common.cpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#define XSTR(x) STR(x)
+#define STR(x) #x
+
```
Comment (slackito) [inline lldb/test/API/functionalities/breakpoint/same_cu_name/common.cpp:3]: These stringification macros from the original bug repro are no longer needed here.

File: lldb/test/API/functionalities/breakpoint/same_cu_name/common.cpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace NAMESPACE {
+  static int g_value = 0;
+void DoSomeStuff() {
```
Comment (slackito) [inline lldb/test/API/functionalities/breakpoint/same_cu_name/common.cpp:7]: Nit: inconsistent indentation.

**Review by slackito — APPROVED** [review]
Thanks for the fix!

Looks good overall, just a couple of nits. I'm not familiar with this part of the code so feel free to wait for any of the other reviewers if you want someone to actually check your logic changes.

**Review by labath — APPROVED** [review]

**Comment by jimingham** [comment]
This patch seems to fail on x86_64 Linux, but I don't understand how this patch could cause the reported failure.  The failure is in a lldb-dap test, here:

File "/var/lib/buildkite-agent/builds/linux-56-59b8f5d88-9dkps-1/llvm-project/github-pull-requests/lldb/test/API/tools/lldb-dap/evaluate/TestDAP_evaluate.py", line 104, in run_test_evaluate_expressions

Comparing the log to a successful one, all the breakpoints got set on the same lines.  Then we ran and are stopped at the first breakpoint - which parallels a successful test.   We also passed a bunch of self.assertEvaluate's to print various variables.  Then we go to run the "var" command in "repl" mode:

```
        if context == "repl":
            # In the repl context expressions may be interpreted as lldb
            # commands since no variables have the same name as the command.
            self.assertEvaluate("var", r"\(lldb\) var\n.*")

```

and get no reply.  Then we try to kill the DAP process, and that also seems to fail.

I can't reproduce this locally, and I can't see how this failure could result from this patch...

I'm going to apply this patch and keep an eye on the x86_64 bots to see if this was just a glitch or not.

**Comment by DavidSpickett** [comment]
This is failing on Windows on Arm, I'm looking into it now.

**Comment by jimingham** [comment]
Thanks for figuring that out!

**Pull request merged**